Adding check for step size=0 in unfold backward to avoid divide by 0 …

mansiag05 opened the pull request:
…or FPE.

Fixes #142462 

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01 @malfet

mansiag05 committed:
Adding check for step size=0 in unfold backward to avoid divide by 0 or FPE.

mansiag05 committed:
Adding check for step size=0 in unfold backward to avoid divide by 0 or FPE for cpu kernel.

isuruf commented on aten/src/ATen/native/cpu/UnfoldBackwardKernel.cpp:
```diff
@@ -115,6 +115,7 @@ void unfold_backward_cpu_kernel(
   int64_t size,
   int64_t step
 ) {
+  TORCH_CHECK_VALUE(step > 0, "step is ", step, " but must be > 0");
```
This check should be added to `aten/src/ATen/native/UnfoldBackward.cpp` so that all devices use it.

mansiag05 replied:
The FPE is occurring only on CPU. For CUDA the check is already handled. 
Hence the  check was added to CPU only.

mansiag05 committed:
Move unfold_backward step validation to common code

Move TORCH_CHECK_VALUE for step > 0 from CPU kernel to shared
unfold_backward function to ensure all backends validate the
parameter.

mansiag05 committed:
Ensuring the unfold_backward op test runs on all backend

isuruf commented:
@pytorchbot merge

mansiag05 committed:
Fixing the Lint error.
